Containerfile

```diff
@@ -181,8 +181,10 @@ ARG NODE_MAJOR
 RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
   --mount=type=cache,target=/var/lib/apt,sharing=locked \
   --mount=type=tmpfs,target=/var/log \
-  curl -sL https://deb.nodesource.com/setup_$NODE_MAJOR.x | bash - && \
-  DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends \
+  mkdir -p -- /etc/apt/keyrings \
+  curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg \
+  echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list \
+  DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install -yq --no-install-recommends \
     nodejs
```

Comment:
While `NODE_MAJOR` is declared in docker-compose, and is the reason why build automation fail, I've extensively test this locally and found that it will fall back to the version already installed in base. Let me know what you thin k @klardotsh 

```
[5/5] STEP 7/15: RUN --mount=type=cache,target=/var/cache/apt,sharing=locked   --mount=type=cache,target=/var/lib/apt,sharing=locked   --mount=type=tmpfs,target=/var/log   mkdir -p -- /etc/apt/keyrings   curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg   echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list   DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install -yq --no-install-recommends     nodejs
usage: gpg [options] --dearmor [file]
Reading package lists...
Building dependency tree...
Reading state information...
nodejs is already the newest version (16.20.2-deb-1nodesource1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

Reply:
I also wonder if this whole section is redundant, since the base image already has nodejs installed. I still keep it around, thinking that `--mount` portion might still provide some kind of optimization.

Reply:
Approving on a "well if we find out later that this whole section is indeed redundant, we can always remove it then" grounds, seems reasonable to me and I'd rather unblock the overall process.